<a href="https://cognitiveclass.ai"><img src = "https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork/labs/Module%202/images/IDSNlogo.png" width = 400> </a>

<h1 align=center><font size = 5>From Requirements to Collection</font></h1>


## Introduction

In this lab, we will continue learning about the data science methodology, and focus on the **Data Requirements** and the **Data Collection** stages.

Estaimted time needed: **15** minutes

## Objectives

After complting this lab you will be able to:

-   Understand Data Requirements
-   Explore the stages in Data Collection


## Table of Contents

<div class="alert alert-block alert-info" style="margin-top: 20px">
    
1. [Data Requirements](#0)<br>
2. [Data Collection](#2)<br>
</div>
<hr>


# Data Requirements <a id="0"></a>


<img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork/labs/Module%202/images/lab2_fig1_flowchart_data_requirements.png" width=500>


In the videos, we learned that the chosen analytic approach determines the data requirements. Specifically, the analytic methods to be used require certain data content, formats and representations, guided by domain knowledge.


In the **From Problem to Approach Lab**, we determined that automating the process of determining the cuisine of a given recipe or dish is potentially possible using the ingredients of the recipe or the dish. In order to build a model, we need extensive data of different cuisines and recipes.


Identifying the required data fulfills the data requirements stage of the data science methodology.

* * *


# Data Collection <a id="2"></a>


<img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork/labs/Module%202/images/lab2_fig2_flowchart_data_collection.png" width=500>


In the initial data collection stage, data scientists identify and gather the available data resources. These can be in the form of structured, unstructured, and even semi-structured data relevant to the problem domain.


#### Web Scraping of Online Food Recipes

A researcher named Yong-Yeol Ahn scraped tens of thousands of food recipes (cuisines and ingredients) from three different websites, namely:


<img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork/labs/Module%202/images/lab2_fig3_allrecipes.png" width=500>
<div align="center">
www.allrecipes.com
</div>
<br/><br/>
<img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork/labs/Module%202/images/lab2_fig4_epicurious.png" width=500>
<div align="center">
www.epicurious.com
</div>
<br/><br/>
<img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork/labs/Module%202/images/lab2_fig5_menupan.png" width=500>
<div align="center">
www.menupan.com
</div>
<br/><br/>


For more information on Yong-Yeol Ahn and his research, you can read his paper on [Flavor Network and the Principles of Food Pairing](http://yongyeol.com/papers/ahn-flavornet-2011.pdf?utm_email=Email&utm_source=Nurture&utm_content=000026UJ&utm_term=10006555&utm_campaign=PLACEHOLDER&utm_id=SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork-20083987).


Luckily, we will not need to carry out any data collection as the data that we need to meet the goal defined in the business understanding stage is readily available.


#### We have already acquired the data and placed it on an IBM server. Let's download the data and take a look at it.


<h4 style="color: red";> Important note: Please note that you are not expected to know how to program in R. The following code is meant to illustrate the stage of data collection, so it is totally fine if you do not understand the individual lines of code. We have a full course on programming in R, <a href="http://cocl.us/RP0101EN_DS0103EN_LAB2_R">R101</a>, so please feel free to complete the course if you are interested in learning how to program in R.


### Using this notebook:

To run any of the following cells of code, you can type **Shift + Enter** to excute the code in a cell.


Get the version of R installed.


In [7]:
import pandas as pd
import requests as rq

Download the data from the IBM server.


In [30]:
recipes='https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork/labs/Module%202/recipes.csv'

Read the data into an R dataframe and name it **recipes**.


In [31]:
df = pd.read_csv(recipes)


In [37]:
df.to_csv(r'C:\Users\Maha Al-Khater\Documents\GitHub\resipes.csv', index = False)

Show the first few rows.


In [33]:
df.head(10)

country almond angelica anise anise_seed apple apple_brandy apricot  \
0  Vietnamese     No       No    No         No    No           No      No   
1  Vietnamese     No       No    No         No    No           No      No   
2  Vietnamese     No       No    No         No    No           No      No   
3  Vietnamese     No       No    No         No    No           No      No   
4  Vietnamese     No       No    No         No    No           No      No   
5  Vietnamese     No       No    No         No    No           No      No   
6  Vietnamese     No       No    No         No    No           No      No   
7  Vietnamese     No       No    No         No    No           No      No   
8  Vietnamese     No       No    No         No    No           No      No   
9  Vietnamese     No       No    No         No    No           No      No   

  armagnac artemisia  ... whiskey white_bread white_wine  \
0       No        No  ...      No          No         No   
1       No        No  ...      No          No         No   
2       No        No  ...      No          No         No   
3       No        No  ...      No          No         No   
4       No        No  ...      No          No         No   
5       No        No  ...      No          No         No   
6       No        No  ...      No          No         No   
7       No        No  ...      No          No         No   
8       No        No  ...      No          No         No   
9       No        No  ...      No          No         No   

  whole_grain_wheat_flour wine wood yam yeast yogurt zucchini  
0                      No   No   No  No    No     No       No  
1                      No   No   No  No    No     No       No  
2                      No   No   No  No    No     No       No  
3                      No   No   No  No    No     No       No  
4                      No   No   No  No    No     No       No  
5                      No   No   No  No    No     No       No  
6                      No   No   No  No    No     No       No  
7                      No   No   No  No    No     No       No  
8                      No   No   No  No    No     No       No  
9                      No   No   No  No    No     No       No  

[10 rows x 384 columns]

Get the dimensions of the dataframe.


In [35]:
df.shape

(57691, 384)

So our dataset consists of 57,691 recipes. Each row represents a recipe, and for each recipe, the corresponding cuisine is documented as well as whether 384 ingredients exist in the recipe or not, beginning with almond and ending with zucchini.

* * *


Now that the data collection stage is complete, data scientists typically use descriptive statistics and visualization techniques to better understand the data and get acquainted with it. Data scientists, essentially, explore the data to:

-   understand its content,
-   assess its quality,
-   discover any interesting preliminary insights, and,
-   determine whether additional data is necessary to fill any gaps in the data.


### Thank you for completing this lab!

This notebook is part of the free course on **Cognitive Class** called _Data Science Methodology_. If you accessed this notebook outside the course, you can take this free self-paced course, online by clicking [here](http://cocl.us/DS0103EN_LAB2_R).

## Author

<a href="https://www.linkedin.com/in/aklson/" target="_blank">Alex Aklson</a>

## Change Log

| Date (YYYY-MM-DD) | Version | Changed By | Change Description                 |
| ----------------- | ------- | ---------- | ---------------------------------- |
| 2021-04-06        | 2.1     | Malika     | Updated lab link                   |
| 2020-09-16        | 2.0     | Simran     | Moved lab to course repo in GitLab |

<hr>

## <h3 align="center"> © IBM Corporation 2020. All rights reserved. <h3/>
